### DISCUSSION TITLE
Connect Emqx to zilla for publish messages to kafka

### DISCUSSION THREAD

**Discussion by sfo-pourya** [aklivity/zilla#1344] [Q&A]
Hi everybody,
I use this [https://github.com/aklivity/zilla-docs/tree/main/src/tutorials/mqtt](tutorial) for use zilla to publish my mqtt message to kafka.
it works with eclipse-mosquitto as mentioned in tutorial but when i want use emqx or mqqtx for connect and publish message.
there is a problem which they can't connect.
any idea?
thanks

**Comment by bmaidics** [comment]
Hi @sfo-pourya . I just tested the example and works for me. Can you elaborate how you try to connect using MQTTX? It was tested with both emqx and mqttx. Don't forget to change the default 1883 port to 7183 in the client settings, as Zilla is configured to be listening on that port. If you want to, you can easily change that port to the desired one in the zilla.yaml.
